org.palladiosimulator.analyzer.slingshot.behavior.spd/src/org/palladiosimulator/analyzer/slingshot/behavior/spd/interpreter/entity/aggregator/AbstractWindowAggregation.java

```diff
@@ -0,0 +1,25 @@
+package org.palladiosimulator.analyzer.slingshot.behavior.spd.interpreter.entity.aggregator;
+
+public abstract non-sealed class AbstractWindowAggregation implements WindowAggregation {
+
+	public AbstractWindowAggregation() {
+		// TODO Auto-generated constructor stub
+	}
+
+	@Override
+	public double getCurrentValue() throws AssertionError {
+		if (!isEmittable())
+			throw new AssertionError(isEmittable());
```

Comment:
dont throw assertion errors, they are for `assert` only as far as i know? go for illegal state, or something like that.